content/guides/caching.md

```diff
@@ -4,284 +4,356 @@ sort: 11
 
-For an even more optimized setup:
+This guide focuses on the configuration needed to ensure files produced by webpack compilation can remain cached unless their contents has changed.
 
-1. Use compiler stats to get the file names when requiring resources in HTML.
-2. Generate the chunk manifest JSON and inline it into the HTML page before loading resources.
 
+## Output Filenames
 
-## The problem
+A simple way to ensure the browser picks up changed files is by using `output.filename` [substitutions](/configuration/output#output-filename). The `[hash]` substitution can be used to include a build-specific hash in the filename, however it's even better to use the `[chunkhash]` subsitution which includes a chunk-specific hash in the filename.
 
-Each time something needs to be updated in our code, it has to be re-deployed on the server and then re-downloaded by all clients. This is clearly inefficient since fetching resources over the network can be slow. This is why browsers cache static resources.
+Let's get our project set up using the example from [getting started](/guides/getting-started) with the `plugins` from [output management](/guides/output-management), so we don't have to deal with maintaining our `index.html` file manually:
 
-The way it works has a pitfall: If we don’t change filenames of our resources when deploying a new version, the browser might think it hasn’t been updated and client will get a cached version of it.
+__project__
 
-A simple way to tell the browser to download a newer version is to alter the asset’s file name. In a pre-webpack era we used to add a build number to the filenames as a parameter and then increment it:
-
-```bash
-application.js?build=1
-application.css?build=1
+``` diff
+webpack-demo
+|- package.json
+|- webpack.config.js
+|- /dist
+|- /src
+  |- index.js
+|- /node_modules
 ```
 
-It is even easier to do with webpack. Each webpack build generates a unique hash which can be used to compose a filename, by including output [substitutions](/configuration/output#output-filename). The following example config will generate 2 files (1 per entry) with hashes in filenames:
-
-```js
-// webpack.config.js
-const path = require("path");
-
-module.exports = {
-  entry: {
-    vendor: "./src/vendor.js",
-    main: "./src/index.js"
-  },
-  output: {
-    path: path.join(__dirname, "build"),
-    filename: "[name].[hash].js"
-  }
-};
+__webpack.config.js__
+
+``` diff
+  const path = require('path');
+  const CleanWebpackPlugin = require('clean-webpack-plugin');
+  const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+  module.exports = {
+    entry: './src/index.js',
+    plugins: [
+      new CleanWebpackPlugin(['dist']),
+      new HtmlWebpackPlugin({
```

Comment:
I actually think the `HtmlWebpackPlugin` is key for this doc. If not, users have to track bundle names themselves using the manifest and I think that complicates the guide significantly. Plus, it's usage is already covered in `output-management` which comes before this guide.